### PR TITLE
Adds unittests

### PR DESCRIPTION
**Pull request by gilbertk88**
Creation of PHPUnit test files.

### CODE REVIEW

**Comment by vpetersson** [comment]
@gilbertk88 It doesn't appear as the tests ran on Travis.

**Comment by vpetersson** [comment]
Ah - `screenly-cast/.travis.yml` is the wrong location. You need to merge it with `.travis.yml`.

**Comment by vpetersson** [comment]
@gilbertk88 It appears as there are still two `.travis.yaml` files.

File: screenly-cast/bin/install-wp-tests.sh [hunk 1]
```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env bash
+
+if [ $# -lt 3 ]; then
+	echo "usage: $0 <db-name> <db-user> <db-pass> [db-host] [wp-version] [skip-database-creation]"
+	exit 1
+fi
+
+DB_NAME=$1
```
Comment (vpetersson) [inline screenly-cast/bin/install-wp-tests.sh]: All variables needs to be quoted (unless `${...}`).

**Comment by rusko124** [comment]
@gilbertk88 the code as a whole retains solid principles, but I think it is necessary to get rid of problems (such as extra spaces, there is no empty line at the end, there is not single style of comments). But in general, I think that the code is good.

File: screenly-cast/bin/install-wp-tests.sh (hunk 1 above)
Comment (gilbertk88) [reply screenly-cast/bin/install-wp-tests.sh]: Okay.

**Comment by gilbertk88** [comment]
Thanks @rusko124 , let me get to it.

File: screenly-cast/bin/install-wp-tests.sh (hunk 1 above)
Comment (gilbertk88) [reply screenly-cast/bin/install-wp-tests.sh]: @vpetersson do the files look good? Have you had a chance to run a test?

Comment (vpetersson) [reply screenly-cast/bin/install-wp-tests.sh]: They should automatically run when a PR is opened, but it doesn't fire properly for some reason.

**Comment by vpetersson** [comment]
@gilbertk88 been a while since i touched this, but i realized that you also need to: 

 * Bump up the version 
 * Populate the [change log](https://github.com/Screenly/Screenly-Cast-for-WordPress/blob/master/screenly-cast/readme.txt)

File: .travis.yml [hunk 2]
```diff
@@ -11,7 +11,6 @@ before_install:
   - sudo apt-get install -qq -y rsync subversion
 branches:
     only:
-        - master
```
Comment (vpetersson) [inline .travis.yml]: @gilbertk88 You can't remove this as this is needed for the deployment.

**Comment by gilbertk88** [comment]
@vpetersson  I see there is a placeholder on the plugin version area, does travis use this?

File: .travis.yml [hunk 3]
```diff
@@ -1,9 +1,71 @@
...
+    if [[ ! -z "$WP_VERSION" ]] ; then
+      phpunit
+      WP_MULTISITE=1 phpunit
+    fi
+  - |
+    if [[ "$WP_TRAVISCI" == "phpcs" ]] ; then
+      phpcs
+    fi
+    
+  - ./bin/push_to_wordpress_org.sh
```
Comment (vpetersson) [inline .travis.yml]: Let's change this to:
```
- test $TRAVIS_BRANCH = "master" && ./bin/push_to_wordpress_org.sh
```

And also remove the branch filter. That way we will run the tests but only deploy when on master.

**Comment by vpetersson** [comment]
@gilbertk88 Almost there - just one minor thing to go.

File: .travis.yml [hunk 4]
```diff
@@ -10,9 +10,8 @@ before_install:
   - sudo apt-get update -qq
   - sudo apt-get install -qq -y rsync subversion
 branches:
```
Comment (vpetersson) [inline .travis.yml]: No - remove the entire 'branches' block.

**Comment by vpetersson** [comment]
Actually @gilbertk88 - the build fails with `Could not parse .travis.yml`

**Pull request merged**